tests/main/kernel-snap-refresh-on-core/task.yaml

```diff
@@ -0,0 +1,99 @@
+summary: Check that the kernel snap can be refreshed on a core device
+
+systems: [ubuntu-core-16-64]
+
+details: |
+    This test checks that the kernel snap can be refreshed from an installed
+    revision to a new one. It expects to find a new snap revision in the
+    channel pointed by the NEW_CORE_CHANNEL env var.
+
+#manual: true
+
+environment:
+    KERNEL_CHANNEL: stable
+    NEW_KERNEL_CHANNEL: edge
+
+restore: |
+    rm -f prevBoot nextBoot prevKernelSignature nextKernelSiganture
+        
+prepare: |
+    snap install test-snapd-tools
+
+execute: |
+    wait_core_post_boot() {
+        # booted
+        while [ "$(bootenv snap_mode)" != "" ]; do
+            sleep 1
+        done
+    }
+
+    . $TESTSLIB/boot.sh
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        # ensure we have a good starting place
+    
+        # sanity
+        test-snapd-tools.echo hello | MATCH hello
+
+        # go to known good starting place
+        snap install pc-kernel --${KERNEL_CHANNEL}
+        REBOOT
+    elif [ "$SPREAD_REBOOT" = 1 ]; then
+       # from our good starting place we refresh
+        wait_core_post_boot
+        
+        # save current core revision
+        snap list | awk "/^pc-kernel / {print(\$3)}" > prevBoot
+        cat /proc/version_signature > prevKernelSignature
+
+        # refresh
+        snap refresh pc-kernel --${NEW_KERNEL_CHANNEL}
+
+        # check boot env vars
+        snap list | awk "/^pc-kernel / {print(\$3)}" > nextBoot
+
+        test "$(bootenv snap_kernel)" = "pc-kernel_$(cat prevBoot).snap"
+        test "$(bootenv snap_try_kernel)" = "pc-kernel_$(cat nextBoot).snap"
+
+        # test-snapd-tools works
+        test-snapd-tools.echo hello | MATCH hello
+
+        REBOOT
+    elif [  "$SPREAD_REBOOT" = 2 ]; then
+        # after refresh to NEW_CHANNEL
+        wait_core_post_boot
+
+        # check boot env vars
+        test "$(bootenv snap_kernel)" = "pc-kernel_$(cat nextBoot).snap"
+        test "$(bootenv snap_try_kernel)" = ""
+
+        cat /proc/version_signature > nextKernelSignature
+
+        if diff -u prevKernelSignature nextKernelSignature; then
+            echo "kernel in edge / stable are the same"
+            exit 1
```

Comment:
Should this be a exit one? Can we somehow skip the test. I worry that this will cause failures in the field at the wost possible moment sometime in the future, when we want to release.